src/path.cr

```diff
@@ -564,15 +564,27 @@ struct Path
 
     name = @name
 
-    if name == "~"
-      name = home.to_kind(@kind).normalize.to_s
-    elsif name.starts_with?("~/")
-      name = home.to_kind(@kind).normalize.join(name.byte_slice(2, name.bytesize - 2)).to_s
+    home = case home
+           when Path, String
```

Comment:
`when Path` should be `home`, and `when String` should be `Path.new(string)`, surely?